tests/source-tree/test_source_tree.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2019-2024 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+
+import logging
+import tempfile
+import unittest
+import warnings
+
+from codebasin.file_parser import FileParser
+from codebasin.preprocessor import CodeNode, DirectiveNode, FileNode
+
+
+class TestSourceTree(unittest.TestCase):
+    """
+    Test SourceTree class.
+    """
+
+    def setUp(self):
+        logging.getLogger("codebasin").disabled = False
+        warnings.simplefilter("ignore", ResourceWarning)
+
+    def test_walk(self):
+        """Check that walk() visits nodes in the expected order"""
+
+        # TODO: Revisit this when SourceTree can be built without a file.
+        with tempfile.NamedTemporaryFile(
+            mode="w",
+            delete_on_close=False,
+            suffix=".cpp",
+        ) as f:
+            source = """
+                #if defined(FOO)
+                void foo();
+                #elif defined(BAR)
+                void bar();
+                #else
+                void baz();
+                #endif
+
+                void qux();
+                """
+            f.write(source)
+            f.close()
```

Comment:
Do you need this if you're in a file context?

Reply:
I'm not sure, but I believe that in this case the answer might be yes.  Although it's not at all obvious, the construction of `FileParser` a bit lower down actually opens the file again.  I wanted to avoid any potential weirdness from opening the same file twice, which is why I closed it first.